include/base/libmesh.h

```diff
@@ -98,9 +98,9 @@ class LibMeshInit
    * parameter to use a user-specified MPI communicator.
    */
   LibMeshInit(int argc, const char * const * argv,
-              MPI_Comm COMM_WORLD_IN=MPI_COMM_WORLD);
+              MPI_Comm COMM_WORLD_IN=MPI_COMM_WORLD, int n_threads=-1);
 #else
-  LibMeshInit(int argc, const char * const * argv);
+  LibMeshInit(int argc, const char * const * argv, int n_threads=-1);
```

Comment:
So ... this doesn't work well.  I should have thought about this harder ahead of time, but the combination of this new argument and the previous optional-argument-in-some-configs means that user code can't be made MPI-agnostic anymore if it's also setting n_threads, because the position of the n_threads argument isn't MPI-agnostic.

Maybe what we need to do is simply finally make that optional-argument-in-some-configs into an optional-argument in every config, just using the TIMPI::communicator shim instead of MPI_Comm.  Let me hijack your PR branch and I'll try that.

Reply:
Ah, yeah I can see how that would be a problem. We had some MPI-dependent code on our side anyway, so it fit in pretty well for us but that definitely won't always be the case.

Thanks for digging into it further!